src/Hl7.Fhir.Base/Rest/BaseFhirClient.cs

```diff
@@ -603,7 +603,7 @@ public virtual void Delete(string resourceType, SearchParams condition)
 
 
             var tx = new TransactionBuilder(Endpoint);
-            var resourceType = Inspector.GetFhirTypeNameForType(typeof(TResource));
+            var resourceType = Inspector.GetFhirTypeNameForType(typeof(TResource)) ?? throw new ArgumentException($"{typeof(TResource).Name} is not recognized as a valid FHIR resource.");
```

Comment:
there is a private method `typeNameOrDie` that does this - other methods use this private method to provide a consistent exception.

Reply:
Ah ok. I was not aware of this. It was a proposal of Visual Studio to do this, and I agreed...